### PR TITLE
FIX: CSV parser bot preserves original line ending

### LINKED ISSUE #1597: CSV line recovery forces Windows line endings
The `recover_line_csv()` function is joining data with `\r\n`, redgardless of the source file newlines:
https://github.com/certtools/intelmq/blob/a2d20df6fd4fa0386fe79e66156537336faf92b0/intelmq/lib/bot.py#L1042

This means that bot test are failing even though the parsing succeeded, because the `raw` values do not match: there is a new CR byte. As a workaround, bot tests are replacing newlines in their test files to account for that, which is not ideal solution.
https://github.com/certtools/intelmq/blob/f7a5c7adb9e6a3b8a65c6e453195fd914fe80c20/intelmq/tests/bots/parsers/generic/test_parser_csv_data_type.py#L20

### PR DESCRIPTION
**Pull request by kamil-certat**
CSV parser bot recognizes the original line ending
in the input report and tries to keep it in the
output. If input has multiple different line endings,
the \r\n is used. This is what Python csv module uses
by default.

Previously, line filtering was handled in a few places,
in case of CSV data it was before creating the handler object
and required rewriting the whole CSV data, with hardcoded line
endings. For proper handling line ending, this the filtering was
moved inside the handler and now it does not depend on any line
ending.

In addition, the filtering code was extracted to remove code
duplication and the CSV dict line parser was updated to use
the original line ending.

Closes #1597

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -89,7 +90,9 @@ CHANGELOG
   - Added amplification factor to UDP scan reports (PR#2238 by elsif2).
   - Added version and build_date to "Vulnerable-HTTP" report (PR#2238 by elsif2).
 - `intelmq.bots.parsers.cymru.parser_cap_program`: The parser mapped the hostname into `source.fqdn` which is not allowed by the IntelMQ Data Format. Added a check (PR#2215 by Sebastian Waldbauer, fixes #2169)
-- `intelmq.bots.parsers.generic.parser_csv`: Use RewindableFileHandle to use the original current line for line recovery (PR#2192 by Sebastian Wagner).
+- `intelmq.bots.parsers.generic.parser_csv`:
+  - Use RewindableFileHandle to use the original current line for line recovery (PR#2192 by Sebastian Wagner).
+  - Recovering CSV lines preserves the original line ending (fix [#1597](https://github.com/certtools/intelmq/issues/1597))
```
Comment (sebix) [inline CHANGELOG.md]:
```suggestion
  - Recovering CSV lines preserves the original line ending (PR#2280 by Kamil Mankowski, fixes [#1597](https://github.com/certtools/intelmq/issues/1597))
```

**Review by sebix — APPROVED** [review]
Looks good, also well documented (inline)

**Comment by sebix** [comment]
The failing test is
```
[2022-12-21T16:58:23.155Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2/dist/codecov' failed with exit code 255
0s
```
which I can't override

**Comment by kamil-certat** [comment]
@sebix Thanks for the review and disabling codecov, I've fixed the merge conflict :)

**Pull request merged**